config/default/namespace.yaml

```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  # This *looks* stupid, but we're about to put the `etcd-cluster-operator-` prefix on here.
+  name: system
```

Comment:
😱 it suggests a failure in a configuration tool when it simultaneously
a) adds indirection & complexity
b) makes config more confusing to read 

maybe not for this PR, theres other cleanup needed in this directory, but what do you think about dropping the resource name prefix & maybe even the namespace override? the mutability of names here is really difficult to keep track on and could get very messy

Reply:
Definitely defer this until another PR. Maybe discuss this offline?